bulk_grades/api.py

```diff
@@ -203,13 +207,20 @@ def get_unique_path(self):
         """
         return self.course_id
 
-    def _get_graded_subsections(self, course_id):
+    def _get_graded_subsections(self, course_id, filter_subsection=None, filter_assignment_type=None):
         """
         Return list of graded subsections.
+
+        If filter_subsection (string, a block usage id) is set, return only that subsection.
+        If filter_assignment_type (string) is set, return only subsections of the appropriate type.
         """
         subsections = {}
         for subsection in grades_api.graded_subsections_for_course_id(course_id):
-            short_block_id = subsection.location.block_id[:8]
+            block_id = text_type(subsection.location.block_id)
+            if ((filter_subsection and block_id != UsageKey.from_string(filter_subsection).block_id)
```

Comment:
If `filter_subsection` needs to be converted to a `UsageKey` (not sure why it would), then why not do that outside the loop, instead of with every iteration?